metadata-ingestion/src/datahub/ingestion/source/hex/datahub.py

```diff
@@ -0,0 +1,385 @@
+import json
+import logging
+import re
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import Any, Dict, Generator, List, Optional
+
+from pydantic import BaseModel
+
+from datahub.ingestion.api.source import SourceReport
+from datahub.ingestion.source.hex.constants import (
+    DATAHUB_API_PAGE_SIZE_DEFAULT,
+    HEX_PLATFORM_URN,
+)
+from datahub.metadata._schema_classes import QueryPropertiesClass, QuerySubjectsClass
+from datahub.metadata.urns import DatasetUrn, QueryUrn, SchemaFieldUrn
+from datahub.sdk.main_client import DataHubClient
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class QueryResponse:
+    """This is the public response model for the HexQueryFetcher."""
+
+    urn: QueryUrn
+    hex_project_id: str
+    dataset_subjects: List[DatasetUrn] = field(default_factory=list)
+    schema_field_subjects: List[SchemaFieldUrn] = field(default_factory=list)
+
+
+@dataclass
+class HexQueryFetcherReport(SourceReport):
+    fetched_query_urns: int = 0
+    fetched_query_objects: int = 0
+    filtered_out_queries_missing_metadata: int = 0
```

Comment:
would be useful to include the start_time in the report as well

might also be useful to support an end_time config - mostly for testing purposes

Reply:
addressed in commit cf9c402000dead838c4fd55c0da3be1429a98e56

I considered on using `BaseTimeWindowConfig` but it sets a `bucket` field does not apply in the case of Hex as of today